Optimize the logic to compare the routing entries between ASIC_DB and the prefixes pushed to DUT

stephenxs opened the pull request:
By doing so the time consumed by the logic is reduced significantly for 100000 routing entries:
- IPv4: 20 minutes => 20 seconds
- IPv6: 22 minutes => 2 minutes

The idea is
1. to extract the prefixes from the output of the DUT in order to avoid using json.load for each routing entry
2. to use set instead of list to check difference. `O(n*n)` => `O(n*lg(n))`

Logs before
```
2022-06-06 03:50:10 test_route_perf.test_perf_add_remove_rou L0229 INFO   | IP route utilization before test start: Used: 6414, Available: 186352, Test count: 100000
2022-06-06 04:08:05 test_route_perf.test_perf_add_remove_rou L0244 INFO   | Time to set 100000 ipv4 routes is 69.45 seconds.
2022-06-06 04:12:46 test_route_perf.test_perf_add_remove_rou L0248 INFO   | Time to del 100000 ipv4 routes is 50.76 seconds.
PASSED                                                                                                                                            [ 50%]
2022-06-06 04:13:13 test_route_perf.test_perf_add_remove_rou L0229 INFO   | IP route utilization before test start: Used: 6416, Available: 186352, Test count: 100000
2022-06-06 04:32:59 test_route_perf.test_perf_add_remove_rou L0244 INFO   | Time to set 100000 ipv6 routes is 132.96 seconds.
2022-06-06 04:38:33 test_route_perf.test_perf_add_remove_rou L0248 INFO   | Time to del 100000 ipv6 routes is 52.06 seconds.
PASSED                                                                                                                                            [100%]
```

Logs after
```
2022-06-06 07:25:43 test_route_perf.exec_routes              L0163 INFO   | Route file generated and copied
2022-06-06 07:25:43 test_route_perf.exec_routes              L0180 INFO   | Before pushing route to swssconfig
2022-06-06 07:26:42 test_route_perf.exec_routes              L0186 INFO   | All route entries have been pushed
2022-06-06 07:26:50 test_route_perf.exec_routes              L0203 INFO   | All route entries have been installed in ASIC_DB in 66.122285 seconds
2022-06-06 07:27:04 test_route_perf.test_perf_add_remove_rou L0253 INFO   | Time to set 100000 ipv4 routes is 66.12 seconds.
2022-06-06 07:27:07 test_route_perf.exec_routes              L0163 INFO   | Route file generated and copied
2022-06-06 07:27:16 test_route_perf.exec_routes              L0180 INFO   | Before pushing route to swssconfig
2022-06-06 07:28:08 test_route_perf.exec_routes              L0186 INFO   | All route entries have been pushed
2022-06-06 07:28:09 test_route_perf.exec_routes              L0203 INFO   | All route entries have been installed in ASIC_DB in 52.581642 seconds
2022-06-06 07:28:09 test_route_perf.test_perf_add_remove_rou L0257 INFO   | Time to del 100000 ipv4 routes is 52.58 seconds.
PASSED                                                                                                                                            [ 50%]
2022-06-06 07:28:43 test_route_perf.exec_routes              L0163 INFO   | Route file generated and copied
2022-06-06 07:28:43 test_route_perf.exec_routes              L0180 INFO   | Before pushing route to swssconfig
2022-06-06 07:29:42 test_route_perf.exec_routes              L0186 INFO   | All route entries have been pushed
2022-06-06 07:30:44 test_route_perf.exec_routes              L0203 INFO   | All route entries have been installed in ASIC_DB in 121.220825 seconds
2022-06-06 07:31:54 test_route_perf.test_perf_add_remove_rou L0253 INFO   | Time to set 100000 ipv6 routes is 121.22 seconds.
2022-06-06 07:31:58 test_route_perf.exec_routes              L0163 INFO   | Route file generated and copied
2022-06-06 07:33:01 test_route_perf.exec_routes              L0180 INFO   | Before pushing route to swssconfig
2022-06-06 07:33:54 test_route_perf.exec_routes              L0186 INFO   | All route entries have been pushed
2022-06-06 07:33:54 test_route_perf.exec_routes              L0203 INFO   | All route entries have been installed in ASIC_DB in 53.888981 seconds
2022-06-06 07:33:55 test_route_perf.test_perf_add_remove_rou L0257 INFO   | Time to del 100000 ipv6 routes is 53.89 seconds.
PASSED                                                                                                                                            [100%]
```

Signed-off-by: Stephen Sun <stephens@nvidia.com>

<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/Azure/SONiC/blob/gh-pages/CONTRIBUTING.md

Please provide following information to help code review process a bit easier:
-->
### Description of PR
<!--
- Please include a summary of the change and which issue is fixed.
- Please also include relevant motivation and context. Where should reviewer start? background context?
- List any dependencies that are required for this change.
-->

Summary:
Fixes # (issue)

### Type of change

<!--
- Fill x for your type of change.
- e.g.
- [x] Bug fix
-->

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)


### Back port request
- [ ] 201911
- [ ] 202012

### Approach
#### What is the motivation for this PR?

#### How did you do it?

#### How did you verify/test it?

#### Any platform specific information?

#### Supported testbed topology if it's a new test case?

### Documentation
<!--
(If it's a new feature, new test case)
Did you update documentation/Wiki relevant to your implementation?
Link to the wiki page?
-->

keboliu commented on tests/platform/test_reboot.py:
```diff
@@ -131,8 +131,14 @@ def reboot_and_check(localhost, dut, interfaces, reboot_type=REBOOT_TYPE_COLD, r
 
-    logging.info("Check reboot cause")
-    check_reboot_cause(dut, reboot_cause)
+    if dut.facts["hwsku"] in sku_supporting_reboot_cause_test:
```
is sku_supporting_reboot_cause_test only for local or also for the community?

stephenxs replied:
fixed.

stephenxs commented:
Community PR [[test_reboot]Wait a few seconds before checking the reboot-cause #1367](https://github.com/Azure/sonic-mgmt/pull/1367) opened.

stephenxs committed:
Optimize the logic to compare the routing entries between ASIC_DB and the prefixes pushed to DUT

By doing so the time consumed by the logic is reduced significantly:
- IPv4: 20 minutes => 20 seconds
- IPv6: 22 minutes => 2 minutes

The idea is
1. to extract the prefixes from the output of the DUT in order to avoid using json.load for each routing entry
2. to use set instead of list to check difference. O(n*n) => O(n*lg(n))

Signed-off-by: Stephen Sun <stephens@nvidia.com>
